examples/materials.rs

```diff
@@ -14,23 +14,18 @@ fn main() {
 
     let geometry = three::Geometry::cylinder(1.0, 2.0, 2.0, 5);
     let mut materials = vec![
-        three::Material::MeshBasic { color: 0xffffff, map: None, wireframe: false },
-        three::Material::MeshLambert { color: 0xffffff, flat: true },
-        three::Material::MeshLambert { color: 0xffffff, flat: false },
-        three::Material::MeshPhong { color: 0xffffff, glossiness: 80.0 },
-        three::Material::MeshPbr {
-            base_color_factor: [0.2, 0.2, 0.2, 1.0],
-            metallic_roughness: [0.5, 0.5],
-            occlusion_strength: 0.2,
-            emissive_factor: [0.0, 0.0, 0.0],
-            normal_scale: 1.0,
-            base_color_map: None,
-            normal_map: None,
-            emissive_map: None,
-            metallic_roughness_map: None,
-            occlusion_map: None,
-        },
+        three::Material::mesh_basic().color(0xffffff).build(),
```

Comment:
I think we should go deeper: user says `three::Material::mesh()` then builds some properties, and then `build` succeeds if Three is able to map those properties to enumeration types we have. E.g. one day we merge lambertian and basic shaders (just a non-realistic example, bear with me), and users will not need to do anything, since it's the `build` implementation logic that only going to change.